File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/TickEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BotState;
import dev.robocode.tankroyale.botapi.BulletState;

import java.util.Collection;

/**
 * Event occurring whenever a new turn in a round has started.
 */
public final class TickEvent extends BotEvent {

    // Current round number.
    private final int roundNumber;

    // Current state of this bot.
    private final BotState botState;

    // Current state of the bullets fired by this bot.
    private final Collection<BulletState> bulletStates;

    // Events occurring in the turn relevant for this bot.
    private final Collection<BotEvent> events;

    /**
     * Initializes a new instance of the TickEvent class.
     *
     * @param turnNumber   is the current turn number in the battle.
     * @param roundNumber  is the current round number in the battle.
     * @param botState     is the current state of this bot.
     * @param bulletStates is the current state of the bullets fired by this bot.
     * @param events       is the events occurring in the turn relevant for this bot.
     */
    public TickEvent(
            int turnNumber,
            int roundNumber,
            BotState botState,
            Collection<BulletState> bulletStates,
            Collection<BotEvent> events) {
        super(turnNumber);
        this.roundNumber = roundNumber;
        this.botState = botState;
        this.bulletStates = bulletStates;
        this.events = events;
    }

    /**
     * Returns the current round number.
     *
     * @return The current round number.
     */
    public int getRoundNumber() {
        return roundNumber;
    }

    /**
     * Returns the current state of this bot.
     *
     * @return The current state of this bot.
     */
    public BotState getBotState() {
        return botState;
    }

    /**
     * Returns the Current state of the bullets fired by this bot.
     *
     * @return The Current state of the bullets fired by this bot.
     */
    public Collection<BulletState> getBulletStates() {
        return bulletStates;
    }

    /**
     * Returns the events that occurred for the bot within the turn.
     *
     * @return The events that occurred for the bot within the turn.
     */
    public Collection<BotEvent> getEvents() {
        return events;
    }
}

File: bot_api_py/events/tick_event.py
from bot_api_py.bot_state import BotState
from bot_api_py.bullet_state import BulletState
from bot_api_py.events.bot_event import BotEvent


class TickEvent(BotEvent):
    """Event occurring whenever a new turn in a round has started."""

    def __init__(self, turn_number: int, round_number: int, bot_state: BotState,
                 bullet_states: list[BulletState], events: list[BotEvent]):
        """Initializes a new instance of the TickEvent class.

        Args:
            turn_number: The current turn number in the battle.
            round_number: The current round number in the battle.
            bot_state: The current state of this bot.
            bullet_states: The current state of the bullets fired by this bot.
            events: The events occurring in the turn relevant for this bot.
        """
        super().__init__(turn_number)
        self.round_number = round_number
        self.bot_state = bot_state
        self.bullet_states = bullet_states
        self.events = events

    def get_round_number(self) -> int:
        """Returns the current round number.

        Returns:
            The current round number.
        """
        return self.round_number

    def get_bot_state(self) -> BotState:
        """Returns the current state of this bot.

        Returns:
            The current state of this bot.
        """
        return self.bot_state

    def get_bullet_states(self) -> list[BulletState]:
        """Returns the current state of the bullets fired by this bot.

        Returns:
            The current state of the bullets fired by this bot.
        """
        return self.bullet_states

    def get_events(self) -> list[BotEvent]:
        """Returns the events that occurred for the bot within the turn.

        Returns:
            The events that occurred for the bot within the turn.
        """
        return self.events